Sources/DiffableDataSource/DiffableTableViewDataSource+Replace.swift

```diff
@@ -10,9 +10,37 @@ import UIKit
 extension DiffableTableViewDataSource {
     // MARK: - Public API
 
-    public func replace(with sections: [DiffableSection<SectionID, RowID>], animatingDifferences: Bool = true) throws {
-        try checkUniqueDiffableSections(sections: sections)
+    public func replace(with sections: [DiffableSection<SectionID, RowID>], animatingDifferences: Bool = true) {
+        checkUniqueDiffableSections(sections: sections) { result in
+            switch result {
+            case let .success(sections):
+                self.unsafeReplace(sections: sections, animatingDifferences: animatingDifferences)
+
+            case let .failure(error):
+                logFault(error: error)
+            }
+        }
+    }
+
+    public func checkUniqueDiffableSections(
+        sections: [DiffableSection<SectionID, RowID>],
+        result: (Result<[DiffableSection<SectionID, RowID>], DiffableError>) -> Void
+    ) {
```

Comment:
I think we should consider changing the function signature to return a result instead:

```
public func checkUniqueDiffableSections(
         sections: [DiffableSection<SectionID, RowID>]
) -> Result<[DiffableSection<SectionID, RowID>], DiffableError> {
```

Reply:
Then, we'd use it like this:

```
public func replace(with sections: [DiffableSection<SectionID, RowID>], animatingDifferences: Bool = true) {
         switch checkUniqueDiffableSections(sections: sections) { 
             case let .success(sections):
                 unsafeReplace(sections: sections, animatingDifferences: animatingDifferences)

             case let .failure(error):
                 logFault(error: error)
             }
         }
     }
```

The same logic would apply to check unique diffable rows.